state/state.go

```diff
@@ -27,3 +34,44 @@ type View interface {
 	NewView(ctx context.Context, changes merkledb.ViewChanges) (merkledb.View, error)
 	GetMerkleRoot(ctx context.Context) (ids.ID, error)
 }
+
+// StateKey holds the name of the key and its permission (Read/Write). By default,
+// initialization of Keys with duplicate key will not work. And to prevent duplicate
+// insertions from overriding the original permissions, use the Add function below.
+// Permission uses bits since transactions are expected to include this encoding
+type Keys map[string]Permission
+
+// The access permission (Read/Write) for each StateKey
+// Specifying RWrite is setting both the Read and Write bit
+type Permission set.Bits
+
+// By default, no permission bits are set
+func NewPermission(permissions ...uint8) Permission {
```

Comment:
I recommend creating a type `PermissionBit` and use that here instead of `uint8`